Packs/Mimecast/Integrations/MimecastEventCollector/MimecastEventCollector.py

```diff
@@ -0,0 +1,567 @@
+AUDIT_LAST_RUN = 'audit_last_run'
+AUDIT_EVENT_DEDUP_LIST = 'audit_event_dedup_list'
+
+LOCAL_LAST_RUN = {SIEM_LAST_RUN: '',
+                  SIEM_EVENTS_FROM_LAST_RUN: '',
+                  AUDIT_LAST_RUN: [],
+                  AUDIT_EVENT_DEDUP_LIST: []
+                  }
+
+AUDIT_EVENT_PAGE_SIZE = 500
+SIEM_LOG_LIMIT = 350
+
+
+class MimecastOptions(IntegrationOptions):
+    app_key: str
+    secret_key: str
+    app_id: str
+    access_key: str
+    base_url: str
+    verify: Optional[bool] = False
+
+
+class MimecastClient(IntegrationEventsClient):
+
+    def __init__(self, request: IntegrationHTTPRequest, options: MimecastOptions):  # pragma: no cover
+        super().__init__(request=request, options=options)
+
+    def prepare_headers(self, uri: str):
+        """
+        Args:
+            uri (str): The uri of the end point
+
+        Returns:
+            The headers part of the request to send to mimecast
+        """
+        # Create variables required for request headers
+        request_id = str(uuid.uuid4())
+        request_date = self.get_hdr_date()
+
+        unsigned_auth_header = '{date}:{req_id}:{uri}:{app_key}'.format(
+            date=request_date,
+            req_id=request_id,
+            uri=uri,
+            app_key=self.options.app_key
+        )
+        hmac_sha1 = hmac.new(
+            base64.b64decode(self.options.secret_key),
+            unsigned_auth_header.encode(),
+            digestmod=hashlib.sha1).digest()
+        sig = base64.encodebytes(hmac_sha1).rstrip()
+        headers = {
+            'Authorization': 'MC ' + self.options.access_key + ':' + sig.decode(),
+            'x-mc-app-id': self.options.app_id,
+            'x-mc-date': request_date,
+            'x-mc-req-id': request_id,
+            'Content-Type': 'application/json'
+        }
+        return headers
+
+    @staticmethod
+    def get_hdr_date():
+        return datetime.utcnow().strftime("%a, %d %b %Y %H:%M:%S UTC")
+
+    def set_request_filter(self, after: Any):   # noqa: F841  # pragma: no cover
+        pass
+
+
+class MimecastGetSiemEvents(IntegrationGetEvents):
+
+    def __init__(self, client: MimecastClient, options: IntegrationOptions):  # pragma: no cover
+        super().__init__(client=client, options=options)
+        self.token: str = ''
+        self.uri: str = '/api/audit/get-siem-logs'
+        self.events_from_prev_run: list = []
+
+    @staticmethod
+    def get_last_run(events: list) -> dict:
+        pass
+
+    def run(self):
+        """
+        Always takes SIEM_LOG_LIMIT amount of events.
+        If the limit is reached, the extra events will be saved in the self.events_from_prev_run.
+        Returns:
+             - stored (list): A list of this run siem events.
+        """
+        self.options.limit = 1 if demisto.command() == 'test-module' else SIEM_LOG_LIMIT
+        stored = []
+        if self.events_from_prev_run and self.options.limit:
```

Comment:
Can `self.options.limit` be empty here?
Seems it always has a value (1 or SIEM_LOG_LIMIT (350))

Reply:
This can be in the Future None or Zero if some one will decide to change the valur SIEM_LOG_LIIMIT